pkg/cmd/server/start/start_allinone.go

```diff
@@ -209,6 +218,33 @@ func (o *AllInOneOptions) Complete() error {
 	o.NodeArgs.NodeName = strings.ToLower(o.NodeArgs.NodeName)
 	o.NodeArgs.MasterCertDir = o.MasterOptions.MasterArgs.ConfigDir.Value()
 
+	// For backward compatibility, enabling node DNS continues to start the master DNS, but the
+	// container DNS server will be the node's. However, if the user has provided an override DNSAddr,
+	// we need to honor the value if the port is not 53 and we do that by disabling node DNS.
+	if !o.IsRunFromConfig() && o.NodeArgs.Components.Enabled(ComponentDNS) {
```

Comment:
I've read this block about 10 times and still don't have all the possible permutations in my head. Is there any way to test the configs that result from various inputs for this?